pyffi/object_models/common.py

```diff
@@ -414,6 +414,35 @@ def get_hash(self, data=None):
         """
         return int(self.get_value()*200)
 
+class HFloat(Float):
+    def read(self, stream, data):
+        """Read value from stream.
+        :param stream: The stream to read from.
+        :type stream: file
+        """
+        self._value = struct.unpack(data._byte_order + 'e',
```

Comment:
This was added in 3.6. I don't know about support for previous versions.
Python half-float is IEEE 754 which might not be the right type for nifs.
Sadly I can't find it (might've been in the dev channel)

Reply:
Ummm, didn't mean to submit review...

Reply:
I think I saw a PR for half float but it was implemented from the ground up, like with mantissa and stuff. I kinda want to avoid that if possible because it makes the whole thing even less maintainable. I guess we have to cut support for ancient versions at some point...

Reply:
I'm mainly thinking that we target 3.7+ going forward.